File: ET3ALEM_SERVER/BusinessEntities/Models/QuestionAttempt/TrueFalseAttempt.cs
namespace BusinessEntities.Models
{
    public class TrueFalseAttempt : QuestionAttempt
    {
        public bool Answer { get; set; }
        public bool IsAnswered { get; set; }

        public override void GradeQuestion()
        {
            IsGraded = true;
            var tfQuestion = QuizQuestion.Question as TrueFalseQuestion;
            double grade = (tfQuestion.Answer == Answer && IsAnswered) ? QuizQuestion.Grade : 0;
            Grade = grade;
        }
    }
}

File: ET3ALEM_CLIENT_WEB/src/app/question/Models/questionAttempt/true-false-attempt.ts
import { QuestionAttempt } from './question-attempt';

export class TrueFalseAttempt extends QuestionAttempt {
  Answer: boolean;
  IsAnswered: boolean;
  constructor(Id, QuizQuestionId, Grade, Answer = false, IsAnswered = false) {
    super(Id, QuizQuestionId, Grade);
    this.Answer = Answer;
    this.IsAnswered = IsAnswered;
  }
}
